crates/bevy_reflect/src/lib.rs

```diff
@@ -833,33 +833,32 @@ mod tests {
     }
 
     #[test]
-    fn from_reflect_should_use_default_container_attribute() {
-        #[derive(Reflect, Eq, PartialEq, Debug)]
+    fn from_reflect_should_require_all_fields() {
```

Comment:
Oh I see now. We're making `FromReflect` fail if any field is missing.

This is a major breakage for any user/crate who relies on `FromReflect`/`ReflectFromReflect` to allow optional fields during deserialization. I know `bevy_proto` makes heavy use of that feature, but I'm sure other reflection-based crates do as well.

I feel like we shouldn't remove that functionality until we introduce a good replacement for it.

Reply:
everything snafu `FromReflect` (or `ReflectFromReflect`) currently does already possible with just `Default` (or `ReflectDefault`) & `Reflect::apply`. `FromReflect` has a pretty explicit meaning which

Reply:
i'll have a look at `bevy_proto`'s use case in the morning but i suspect that i'll consider it a logic error. admittedly my conception isn't reflected in docs AFAIR but i'm happy to change that.

Reply:
Yeah that makes sense to me. Could you add the `ReflectDefault` + `Reflect::apply` flow to the Migration Guide?

Also, a test to verify this flow works could be nice, but not strictly necessary since I'm pretty sure it does.

Reply:
> i'll have a look at `bevy_proto`'s use case in the morning but i suspect that i'll consider it a logic error.

I'm pretty sure the `ReflectDefault` solution would work for those cases. Basically, I need to create `Schematic` implementations of Bevy items, which need a `Schematic::Input` type. And so I was using `#[reflect(Default)]` as a way to get around having to add the `#[reflect(default = "field_specific_default")]` function to nearly every field in order to make it optional haha (`Style` was a killer 😵)